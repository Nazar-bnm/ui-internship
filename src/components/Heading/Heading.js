import React from 'react';
import PropTypes from 'prop-types';

import './Heading.scss';

const CN = 'heading';
const posEnum = {
  right: 'right',
  center: 'center'
};

const Heading = ({ title, position }) => (
  <div className={`${CN}__wrapper`}>
    <div className={`${CN}__line`} />
    <div className={`${CN}__title ${posEnum[position]}`}>
      {title}
    </div>
  </div>
);

Heading.propTypes = {
  title: PropTypes.string,
  position: PropTypes.oneOf(Object.keys(posEnum))
};

Heading.defaultProps = {
  title: 'Default heading',
  position: 'left'
};

export default Heading;
