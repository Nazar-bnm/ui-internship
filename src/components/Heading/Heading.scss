@import '@/scss/_global_styles';

.heading {
  &__wrapper {
    @include flex;
    align-content: center;
    position: relative;
    width: 100%;
    height: 25px;
  }

  &__line {
    width: 100%;
    height: 2px;
    background-color: $alto;
  }

  &__title {
    position: absolute;
    padding: 0 10px;
    left: 0;
    background-color: $white;
    text-transform: uppercase;
  }
}

.right {
  right: 0;
  left: auto;
}

.center {
  left: 50%;
  transform: translate(-50%);
}