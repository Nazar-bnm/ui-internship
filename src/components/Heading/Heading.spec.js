import React from 'react';
import { shallow } from 'enzyme';

import Heading from './Heading';

describe('<Heading />', () => {
  test('should match snapshot', () => {
    let wrapper = shallow(
      <Heading />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
