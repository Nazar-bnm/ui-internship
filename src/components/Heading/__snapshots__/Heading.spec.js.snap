// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should match snapshot 1`] = `
<div
  className="heading__wrapper"
>
  <div
    className="heading__line"
  />
  <div
    className="heading__title center"
  >
    Default title
  </div>
</div>
`;
